### DISCUSSION TITLE
Does anybody have a way to manually get metadata added through Photoprism back into photos?

### DISCUSSION THREAD

**Discussion by shaun-blake** [photoprism/photoprism#1524] [Q&A] [answered]
Hi,

I know that writing metadata into the photos themselves is under consideration.  I'm wondering, does anyone knows of a way to get them back into the files as things are currently?  I'm mostly interested in the title, description, keywords, and labels.

I believe the metadata is currently only (easily) available in the yaml files, is that right?  Does anyone know if it's possible to use the yaml files and exiftool to add those back into the photos or maybe json sidecar files as a manual process?

I'm just trying to avoid putting any work into my photos through Photoprism that I won't be able to use without Photoprism.

Thanks!

**Comment by lastzero** [comment]
There should be a related issue with discussion already. Whenever writing to original files, there is a general risk of corruption especially when files had issues before. So it's something to carefully consider. Also there isn't one set of fields but a ton of it and every app uses them differently. Ideally we can implement write support for a subset of XMP. It's a small software project in itself. Obviously you can use the data we provide via YAML or REST API and write it to your files at your own risk, eg via Exiftool in a shell script.

**Comment by graciousgrey** [comment, marked as answer]
See https://github.com/photoprism/photoprism/issues/402 and https://github.com/photoprism/photoprism/discussions/1092 :)

**Comment by shaun-blake** [comment]
Thanks for the pointers to the existing issue/discussion. As I mentioned, I'm aware the feature to write the metadata is under consideration and that great care needs to be taken. This discussion was to see if anyone already knew of a way or implemented something themselves to do it manually until such a feature exists.  I'm trying to decide between using Photoprism now is worth the effort of writing something myself if nobody else already has, or if I should just wait until (if ever) the feature is implemented. 

I've updated the discussion title to better reflect the question.

**Reply by nagarrido** [reply]
You can do it with a shell script amd exif tool. I've not done it with yaml files, but I do it frequently with json files from google photos (as I'm taking everything away from google). Should not be very different, you could give that a try 😉

**Reply by lastzero** [reply]
It's much easier to write a script for your own needs than for us to provide a general solution that works for everyone and is well documented for a long list of apps / metadata models.

Anyhow, your data is kept in easy to understand, machine readable formats when using PhotoPrism. Once we have the resources, we will make use of that for exporting / syncing. Most users will need a nice UI and comprehensive docs as there are many details to consider.

**Reply by shaun-blake** [reply]
Yeah, I wouldn't expect something like that to be part of the project. Just checking if someone in the community might have already written something or had a method they could share.

**Reply by JiCiT** [reply]
I wasn't able to find anything that does it, so I made one.  See https://github.com/photoprism/photoprism/discussions/1092#discussioncomment-10205965

**Reply by graciousgrey** [reply]
Thank you very much for sharing :) If you like you can add it to https://github.com/photoprism/photoprism-contrib.
